homeassistant/components/wmspro/cover.py

```diff
@@ -46,12 +46,12 @@ class WebControlProAwning(WebControlProGenericEntity, CoverEntity):
     def current_cover_position(self) -> int | None:
         """Return current position of cover."""
         action = self._dest.action(WMS_WebControl_pro_API_actionDescription.AwningDrive)
-        return action["percentage"]
+        return 100 - action["percentage"]
 
     async def async_set_cover_position(self, **kwargs: Any) -> None:
         """Move the cover to a specific position."""
         action = self._dest.action(WMS_WebControl_pro_API_actionDescription.AwningDrive)
-        await action(percentage=kwargs[ATTR_POSITION])
+        await action(percentage=100 - kwargs[ATTR_POSITION])
 
     @property
     def is_closed(self) -> bool | None:
```

Comment:
As a middle-ground-solution, we could make this function return the awnings being closed (now meaning expanded) as soon as the current position in HA methodology is less than 100% (now meaning fully retracted).

Any opinions about this?

Reply:
@joostlek what do you think?